### PR TITLE
[air] Add annotation for Tune module.

### PR DESCRIPTION
**Pull request by xwjiang2010**
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/execution/placement_groups.py [hunk 1]
```diff
@@ -248,7 +248,9 @@ def __repr__(self) -> str:
         )
 
 
+@PublicAPI(stability="alpha")
```
Comment (krfricke) [inline python/ray/tune/execution/placement_groups.py:251]: I think this should be DeveloperAPI or private. Actually, I think I prefer private but lets discuss

File: python/ray/tune/logger/logger.py [hunk 2]
```diff
@@ -201,6 +201,7 @@ def log_trial_end(self, trial: "Trial", failed: bool = False):
                 trial_loggers[trial].close()
 
 
+@PublicAPI(stability="alpha")
```
Comment (krfricke) [inline python/ray/tune/logger/logger.py:204]: Let's make this DeveloperAPI, regular users don't need this

```suggestion
@DeveloperAPI
```

**Review by krfricke — COMMENTED** [review]
Generally looks good, minor requests

**Review by krfricke — APPROVED** [review]
Awesome!

**Review by kouroshHakha — APPROVED** [review]

**Comment by xwjiang2010** [comment]
`python/ray/tune/tests/test_tuner_restore.py` passes locally.

**Pull request merged**